test/unit/models/outbound/pispOTPValidate.model.test.ts

```diff
@@ -0,0 +1,222 @@
+
+  beforeEach(async () => {
+    let subId = 0
+    let handler: NotificationCallback
+
+    modelConfig = {
+      key: 'cache-key',
+      kvs: new KVS(connectionConfig),
+      pubSub: new PubSub(connectionConfig),
+      logger: connectionConfig.logger,
+      thirdpartyRequests: {
+        patchConsentRequests: jest.fn(() => Promise.resolve({ statusCode: 202 })),
+      } as unknown as ThirdpartyRequests,
+      requestProcessingTimeoutSeconds: 3
+    }
+    mocked(modelConfig.pubSub.subscribe).mockImplementationOnce(
+      (_channel: string, cb: NotificationCallback) => {
+        handler = cb
+        return ++subId
+      }
+    )
+
+    mocked(modelConfig.pubSub.publish).mockImplementationOnce(
+      async (channel: string, message: Message) => handler(channel, message, subId)
+    )
+    await modelConfig.kvs.connect()
+    await modelConfig.pubSub.connect()
+  })
+
+  afterEach(async () => {
+    await modelConfig.kvs.disconnect()
+    await modelConfig.pubSub.disconnect()
+  })
+
+  function checkPISPOTPModelLayout (pispOTPmodel: PISPOTPValidateModel, optData?: PISPOTPValidateData) {
+    expect(pispOTPmodel).toBeTruthy()
+    expect(pispOTPmodel.data).toBeDefined()
+    expect(pispOTPmodel.fsm.state).toEqual(optData?.currentState || 'start')
+
+    // check new getters
+    expect(pispOTPmodel.pubSub).toEqual(modelConfig.pubSub)
+    expect(pispOTPmodel.thirdpartyRequests).toEqual(modelConfig.thirdpartyRequests)
+
+    // check is fsm correctly constructed
+    expect(typeof pispOTPmodel.fsm.init).toEqual('function')
+    expect(typeof pispOTPmodel.fsm.validateOTP).toEqual('function')
+
+    // check fsm notification handler
+    expect(typeof pispOTPmodel.onValidateOTP).toEqual('function')
+
+    expect(sortedArray(pispOTPmodel.fsm.allStates())).toEqual([
+      'OTPIsValid',
+      'errored',
+      'none',
+      'start'
+    ])
+    expect(sortedArray(pispOTPmodel.fsm.allTransitions())).toEqual([
+      'error',
+      'init',
+      'validateOTP'
+    ])
+  }
+
+  it('module layout', () => {
+    expect(typeof PISPOTPValidateModel).toEqual('function')
+    expect(typeof create).toEqual('function')
+  })
+
+  describe('Validate OTP with DFSP Phase', () => {
+    let validateData: PISPOTPValidateData
+    const consentPostResponse: tpAPI.Schemas.ConsentsPostRequest = {
+      consentId: "8e34f91d-d078-4077-8263-2c047876fcf6",
+      consentRequestId: "6988c34f-055b-4ed5-b223-b10c8a2e2329",
+      scopes: [{
+          accountId: "some-id",
+          actions: [
+            "accounts.getBalance",
+            "accounts.transfer"
+          ]
+        }
+      ]
+    }
+
+    const genericErrorResponse: fspiopAPI.Schemas.ErrorInformationObject = {
+      errorInformation: {
+        errorCode: '6000',
+        errorDescription: 'Generic thirdparty error'
+      }
+    }
+
+    beforeEach(async () => {
+      validateData = {
+        consentRequestsRequestId: '45789d6b-9423-4876-ab98-a12ff5b54195',
```

Comment:
let define `'45789d6b-9423-4876-ab98-a12ff5b54195'` as independent `const` so it can be reused instead copy/pasting the value around